src/JarConnector.sol

```diff
@@ -57,39 +66,52 @@ contract JarConnector {
         }
     }
 
-    // callable by anyone
-    function ethExit(uint wad, bytes32 ilk_) public {
-        ilk_; // shh compiler wanring
-        ethJoin.exit(address(this), wad);
-    }
-
-    function ethExit() public {
-        ethExit(vat.gem(ilk, address(this)), ilk);
-    }
-
     function getUserScore(bytes32 user) external view returns (uint) {
         if(round == 0) return 0;
 
         uint cdp = uint(user);
+        bytes32 ilk = man.ilks(cdp);
+
+        // Should return 0 score for unsupported ilk
+        if( ! isIlkSupported[ilk]) return 0;
```

Comment:
this optimization can be removed. If user want to vote/draw funds with a CDP that is not enitled to it, there is no reason to make our code more complex in order to save him some gas...

Reply:
As per the discussion, we would need this check to ensure that a user gets the score only for the supported ilks. Otherwise, an attacker can open a CDP with non-supported ilk and claim score.